CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/ApplyColorblindAction.java

```diff
@@ -0,0 +1,169 @@
+import au.gov.asd.tac.constellation.plugins.PluginInteraction;
+import au.gov.asd.tac.constellation.plugins.parameters.PluginParameters;
+import au.gov.asd.tac.constellation.preferences.ApplicationPreferenceKeys;
+import au.gov.asd.tac.constellation.utilities.color.ConstellationColor;
+import java.util.prefs.Preferences;
+import org.openide.awt.ActionID;
+import org.openide.awt.ActionReference;
+import org.openide.awt.ActionRegistration;
+import org.openide.util.NbBundle;
+import org.openide.util.NbPreferences;
+
+/**
+ *
+ * @author centauri0320001
+ */
+
+/*
+ * toolbar options containing set of colourblind actions
+ */
+@ActionID(category = "Display", id = "au.gov.asd.tac.constellation.graph.visual.plugins.colorblind.ApplyColorblindAction")
+@ActionRegistration(displayName = "#CTL_ApplyColorblindAction", iconBase = "au/gov/asd/tac/constellation/graph/visual/plugins/colorblind/resources/colorblind.png", surviveFocusChange = true)
+@ActionReference(path = "Menu/Display", position = 950, separatorBefore = 901)
+@NbBundle.Messages("CTL_ApplyColorblindAction=Apply Selected Colorblind Schema to Graph")
+public final class ApplyColorblindAction extends SimpleAction {
+
+    private static Preferences prefs = NbPreferences.forModule(ApplicationPreferenceKeys.class);
+    //Retrieve colorblind mode selection preference 
+    private static String colorMode = prefs.get(ApplicationPreferenceKeys.COLORBLIND_MODE, ApplicationPreferenceKeys.COLORBLIND_MODE_DEFAULT);
+
+    public ApplyColorblindAction(GraphNode context) {
+        super(context);
+    }
+
+    @Override
+    protected void edit(PluginGraphs graphs, PluginInteraction interaction, PluginParameters parameters) throws InterruptedException, PluginException {
+        final Graph graph = graphs.getGraph();
+        final WritableGraph wg = graph.getWritableGraph("Changing colors", true);
+        try {
+            run(wg);
+        } finally {
+            wg.commit();
+        }
+    }
+
+    public void run(final WritableGraph graph) {
+        final int colorAttr = VisualConcept.VertexAttribute.COLOR.get(graph);
+        prefs = NbPreferences.forModule(ApplicationPreferenceKeys.class);
+        colorMode = prefs.get(ApplicationPreferenceKeys.COLORBLIND_MODE, ApplicationPreferenceKeys.COLORBLIND_MODE_DEFAULT);
+
+        // Color the taxonomies so we can see what's going on.
+        final int typeAttributeId = AnalyticConcept.VertexAttribute.TYPE.get(graph);
+
+        if (typeAttributeId != Graph.NOT_FOUND) {
+
+            final int vertexCount = graph.getVertexCount();
+            for (int vertex = 0; vertex < vertexCount; vertex++) {
+
+                final int vxId = graph.getVertex(vertex);
+                final SchemaVertexType vertexType = graph.getObjectValue(typeAttributeId, vxId);
+
+                if (typeAttributeId != Graph.NOT_FOUND) {
+                    ConstellationColor caseColor;
+
+                    switch (vertexType.getName()) {
```

Comment:
is there a way to be able to write this section in a more generic sense? Noting that Constellation is extendable, people may want to be able to have this function apply for their own custom types beyond just what is available in Core (i.e. this function should work for any schema defined on top of Constellation)